src/pudl/transform/eia860.py

```diff
@@ -277,20 +277,24 @@ def _core_eia860__generators(
         "ferc_qualifying_facility",
     ]
     gens_df = (
-        pd.concat([ge_df, gp_df, gr_df, g_df], sort=True)
+        pd.concat([x for x in [ge_df, gp_df, gr_df, g_df] if not x.empty], sort=True)
```

Comment:
This change is unrelated to performance but fixes the following warning that kept distracting me:

```
/Users/katie/Documents/work/catalyst/pudl/src/pudl/transform/eia860.py:280: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.
  pd.concat([ge_df, gp_df, gr_df, g_df], sort=True)
```